File: src/main/java/src/SistemaGestorTareas/Usuario.java
package SistemaGestorTareas;

import java.util.ArrayList;
import java.util.List;

public class Usuario {
    private String nombre;
    private List<Proyecto> proyectos;

    public Usuario(String nombre) {
        this.nombre = nombre;
        this.proyectos = new ArrayList<>();
    }

    public String getNombre() {
        return nombre;
    }

    public void crearProyecto(String nombreProyecto) {
        Proyecto nuevoProyecto = new Proyecto(nombreProyecto);
        proyectos.add(nuevoProyecto);
        System.out.println("Proyecto '" + nombreProyecto + "' creado por " + nombre + ".");
    }

    public Proyecto obtenerProyecto(String nombreProyecto) {
        for (Proyecto proyecto : proyectos) {
            if (proyecto.getNombre().equalsIgnoreCase(nombreProyecto)) {
                return proyecto;
            }
        }
        return null;
    }

    @Override
    public String toString() {
        return "Usuario: " + nombre;
    }
}

File: src/main/kotlin/SistemaGestorTareas/Usuario.kt
package SistemaGestorTareas

class Usuario(val nombre: String) {
    private val proyectos: MutableList<Proyecto> = ArrayList()

    fun crearProyecto(nombreProyecto: String) {
        val nuevoProyecto = Proyecto(nombreProyecto)
        proyectos.add(nuevoProyecto)
        println("Proyecto '$nombreProyecto' creado por $nombre.")
    }

    fun obtenerProyecto(nombreProyecto: String?): Proyecto? {
        for (proyecto in proyectos) {
            if (proyecto.nombre.equals(nombreProyecto, ignoreCase = true)) {
                return proyecto
            }
        }
        return null
    }

    override fun toString(): String {
        return "Usuario: $nombre"
    }
}